test/acceptance/todo-cli-test.js

```diff
@@ -764,583 +699,741 @@ describe('todo usage', () => {
+      },
+      {
+        name: '.lint-todorc.js todo configuration',
+        isLegacy: false,
+        setTodoConfig: (daysToDecay, daysToDecayByRule) =>
+          project.setLintTodorc('ember-template-lint', daysToDecay, daysToDecayByRule),
+      },
+    ]) {
+      describe(name, () => {
+        it('removes expired todo file if a todo item has expired when running with --clean-todo', async function () {
+          project.setConfig({
+            rules: {
+              'require-button-type': true,
+            },
+          });
 
-      let result = await run(['.', '--update-todo']);
+          project.write({
+            app: {
+              templates: {
+                'require-button-type.hbs': '<button>Check Expiration</button>',
+              },
+            },
+          });
 
-      const todos = [...(await readTodos(project.baseDir)).values()];
+          setTodoConfig({
+            error: 5,
+          });
 
-      expect(result.exitCode).toEqual(0);
+          // generate todo based on existing error
+          await run(['.', '--update-todo'], {
+            // change the date so errorDate is before today
+            env: {
+              TODO_CREATED_DATE: subDays(new Date(), 10).toJSON(),
+            },
+          });
 
-      for (const todo of todos) {
-        expect(differenceInDays(new Date(todo.errorDate), new Date(todo.createdDate))).toEqual(10);
-      }
-    });
+          // run normally and expect the issue to be back in the error state and there to be no todo
+          let result = await run(['.', '--clean-todo']);
 
-    it('should create todos with correct error date set via env var (overrides package.json)', async function () {
-      project.setConfig({
-        rules: {
-          'no-bare-strings': true,
-        },
-      });
-      project.write({
-        app: {
-          templates: {
-            'application.hbs': '<div>Bare strings are bad...</div>',
-          },
-        },
-      });
-      project.writeTodoConfig({
-        error: 10,
-      });
+          let todoDirs = fs.readdirSync(getTodoStorageDirPath(project.baseDir));
 
-      let result = await run(['.', '--update-todo'], {
-        env: {
-          TODO_DAYS_TO_ERROR: '30',
-        },
-      });
+          expect(result.exitCode).toEqual(1);
+          expect(result.stdout).toMatchInlineSnapshot(`
+              "app/templates/require-button-type.hbs
+                1:0  error  All \`<button>\` elements should have a valid \`type\` attribute  require-button-type
 
-      const todos = [...(await readTodos(project.baseDir)).values()];
+              ✖ 1 problems (1 errors, 0 warnings)
+                1 errors and 0 warnings potentially fixable with the \`--fix\` option."
+            `);
+          expect(todoDirs).toHaveLength(0);
+        });
 
-      expect(result.exitCode).toEqual(0);
+        it('should error if daysToDecay.error is less than daysToDecay.warn in package.json', async function () {
```

Comment:
```suggestion
        it('should error if daysToDecay.error is less than daysToDecay.warn in the config', async function () {
```

**question:** since this group of tests loops through legacy todo configuration, package.json todo configuration, and .lint-todorc.js todo configuration, should these tests be updated to refer to "the config" rather than `package.json`?

Reply:
Agreed!

Reply:
Small verbal nit-pick here: I don't want to consider the prior config as "legacy". I'd like to think of it as short hand (I'd expect most folks to use this syntax just to have the same values for all engine types), but if you _need_ the more detailed "engine + rule" specificity you can drop down and do the new syntax.

Reply:
That's fair. We can change this.